src/orders/src/main/java/mushop/orders/config/MonitoringConfiguration.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) 2016, 2020, Oracle and/or its affiliates.  All rights reserved.
+ * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
+ */
+package  mushop.orders.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+
+import com.oracle.bmc.Region;
+import com.oracle.bmc.auth.AuthenticationDetailsProvider;
+import com.oracle.bmc.auth.SimpleAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.StringPrivateKeySupplier;
+
+import com.oracle.bmc.ConfigFileReader;
+import com.oracle.bmc.ConfigFileReader.ConfigFile;
+import com.oracle.bmc.auth.ConfigFileAuthenticationDetailsProvider;
+import com.oracle.bmc.monitoring.MonitoringClient;
+import com.oracle.bmc.monitoring.model.Datapoint;
+import com.oracle.bmc.monitoring.model.MetricDataDetails;
+import com.oracle.bmc.monitoring.model.PostMetricDataDetails;
+import com.oracle.bmc.monitoring.requests.PostMetricDataRequest;
+import com.oracle.bmc.monitoring.responses.PostMetricDataResponse;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+@Configuration
+public class MonitoringConfiguration {
+
+    @Autowired
+    private Environment env;
+    
+    private MonitoringClient monitoringClient = null;
+    private String compartmentId = null;
+    private String namespace = null;
+    private String region = null;
+
+    public MonitoringClient getMonitoringClient() {
+	return monitoringClient;
+    }
+
+    public String getRegion() {
+	return region;
+    }
+
+    public String getNamespace() {
+	return namespace;
+    }
+
+    public String getCompartmentId() {
+	return compartmentId;
+    }
+    
+    public MonitoringClient initConnection() throws Exception {
```

Comment:
This breaks the current usage. Orders should be able to run independently from any manual requirements and of course, the worse, need to create credentials to be used.

Reply:
+1

Reply:
Agreed. BTW, credentials are totally optional, the doc needs to be corrected. If no credentials set, the code just ignores sending the metrics out to OCI monitoring.

Reply:
implementation 'org.springframework.boot:spring-boot-starter-actuator'

All those dependencies were taken from shipping service. Its was a bit of trial and error. Let me check on this.